src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1488,6 +1490,28 @@ private void ExecuteRestartSearchCommand(object sender, ExecutedRoutedEventArgs
                 .PostOnFailure(nameof(PackageManagerControl), nameof(ExecuteRestartSearchCommand));
         }
 
+        private void ExecuteSearchPackageCommand(object sender, ExecutedRoutedEventArgs e)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            var alternatePackageId = e.Parameter as string;
+            if (!string.IsNullOrWhiteSpace(alternatePackageId))
+            {
+                if (_windowSearchHost != null && _windowSearchHost.IsEnabled)
+                {
+                    if (_windowSearchHost.SearchTask != null)
+                    {
+                        // Terminate current search by reporting it as complete
+                        _currentSearchCallback?.ReportComplete(_windowSearchHost.SearchTask, dwResultsFound: 0);
```

Comment:
Should the concept of stopping the old search and starting a new one exist in the Search method so it affects all searches rather than just this command?

Reply:
Actually, documentation says that it's possible to stop a search by calling `windowSearchHost.SearchAsync(null)`.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivswindowsearchhost.searchasync?view=visualstudiosdk-2019#Microsoft_VisualStudio_Shell_Interop_IVsWindowSearchHost_SearchAsync_Microsoft_VisualStudio_Shell_Interop_IVsSearchQuery_

I will test that.

Reply:
Changed to `IVsWindowSearchHost.SearchAsync(IVsSearchQuery:null)`